From fdcc2f36f549a18ea6f587fb9e9bad97f0114c59 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Thu, 28 Apr 2011 18:25:15 +0000
Subject: progress-ui-handler: don't add progress infos to the window twice

We were adding them twice in case there was already one operation
running.

https://bugzilla.gnome.org/show_bug.cgi?id=648857
---
diff --git a/src/nautilus-progress-ui-handler.c b/src/nautilus-progress-ui-handler.c
index 2479682..fe295b2 100644
--- a/src/nautilus-progress-ui-handler.c
+++ b/src/nautilus-progress-ui-handler.c
@@ -315,10 +315,10 @@ handle_new_progress_info (NautilusProgressUIHandler *self,
 			  G_CALLBACK (progress_info_finished_cb), self);
 
 	self->priv->active_infos++;
-	progress_ui_handler_add_to_window (self, info);
 
 	if (self->priv->active_infos == 1) {
 		/* this is the only active operation, present the window */
+		progress_ui_handler_add_to_window (self, info);
 		gtk_window_present (GTK_WINDOW (self->priv->progress_window));
 	} else {
 		if (gtk_widget_get_visible (self->priv->progress_window)) {
--
cgit v0.9
